fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/cursors/ConcatCursor.java

```diff
@@ -218,6 +220,20 @@ private ConcatCursorContinuation(boolean secondCursor, @Nonnull RecordCursorResu
             }
         }
 
+        @Override
+        @Nonnull
+        public ByteString toByteString() {
+            if (isEnd()) {
+                return ByteString.EMPTY;
+            } else {
+                //form bytes exactly once
+                if (cachedByteString == null) {
+                    cachedByteString = continuationFunction.apply(nextResult.getContinuation().toBytes()).toByteString();
```

Comment:
I feel like the `toBytes` method on this continuation should be modified to call `.toByteString` so that `cachedByteString` also gets populated and then doesn't need to be recomputed if someone calls `toByteString` followed by `toBytes`. It also means there's only one place where we need to have logic that calls the continuation function.

I think the other continuations that you overrode could probably follow suit.

Reply:
Fixed it. Please take a look. @alecgrieser